test/functional/test-3p.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+ import {validateSrcPrefix, validateSrcContains} from '../../src/3p';
+
+ describe('3p', () => {
+  it('throw an error if prefix is not https:', () => {
+    expect(() => {
+      validateSrcPrefix('https:', 'http://adserver.adtechus.com');
+    }).to.throw(/Invalid src/);
+  });
+
+  it('should return undefined if source starts with https', () => {
+    expect(
+      validateSrcPrefix('https:', 'https://adserver.adtechus.com')
+    ).to.equal(undefined);
+  });
+
+  it('throw an error if src does not contain addyn', () => {
+    expect(() => {
+      validateSrcContains('/addyn/', 'http://adserver.adtechus.com/');
+    }).to.throw(/Invalid src/);
+  });
+
+  it('should return undefined if source contains /addyn/', () => {
+    expect(
```

Comment:
I'd write this as just the function call. The return value doesn't matter.

You can also do expect(…).to.not.throw I believe